x/vote/types/vote_router.go

```
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/axelarnetwork/axelar-core/x/vote/exported"
+)
+
+// VoteRouter implements a Vote router based on module name.
+type VoteRouter interface {
+	AddHandler(module string, handler exported.VoteHandler) VoteRouter
+	HasHandler(module string) bool
+	GetHandler(module string) exported.VoteHandler
+	Seal()
+}
+
+var _ VoteRouter = (*router)(nil)
+
+type router struct {
+	routes map[string]exported.VoteHandler
+	sealed bool
+}
+
+// NewRouter creates a new Router interface instance
+func NewRouter() VoteRouter {
+	return &router{
+		routes: make(map[string]exported.VoteHandler),
+	}
+}
+
+// Seal prevents additional route handlers from  being added to the router.
+func (r *router) Seal() {
+	r.sealed = true
+}
+
+// AddHandler registers a vote handler for a given path and returns the handler.
+// Panics if the router is sealed, module is an empty string, or if the module has been registered already.
+func (r *router) AddHandler(module string, handler exported.VoteHandler) VoteRouter {
+	if r.sealed {
+		panic("cannot add handler (router sealed)")
+	}
+
+	if module == "" {
+		panic("module name cannot be an empty string")
+	}
+
+	if r.HasHandler(module) {
```

Should we return an error and let the caller panic instead?